### PR TITLE
Update default java version to 8

### PR DESCRIPTION
**Pull request by StefanSpieker**
I'm not sure why the default is still set to **java 6**, so I updated the default to 8 and also updated the comment accordingly.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -880,8 +880,8 @@
     <!-- By default only check remote repositories once per week -->
     <maven.repository.update.freqency>interval:10080</maven.repository.update.freqency>
 
-    <!-- Starting from Jenkins 1.625.x Jenkins supports Java 7 only, hence we may want to bump the dependency in components -->
-    <java.level>6</java.level>
+    <!-- Starting from Jenkins 2.54 Jenkins supports Java 8 only -->
```
Comment (oleg-nenashev) [inline pom.xml:883]: Well, it also supports Java 11 at the moment. I would rephrase or remove the comment entirely

Comment (StefanSpieker) [reply pom.xml]: Thanks, I adjusted the comment 👍

**Review by timja — APPROVED** [review]
